### ISSUE TITLE
Combine multiple PDFs bug?

### ISSUE THREAD

**Issue by oxygen** [Hopding/pdf-lib#30]
Hi, I'm trying to combine multiple similar PDFs.

All 4 input files are nearly identical, except for a text with some unique number. 
All PDFs have a single page.
The output PDF has 4 pages, yet all pages are identical.

```javascript
const PDFLib = require("pdf-lib");

/**
 * Combines multiple PDFs into one.
 * 
 * @param {Buffer[]} arrBuffers 
 */
async function combine(arrBuffers)
{
	if(!arrBuffers.length)
	{
		throw new Error("arrBuffers is empty. Need at least one PDF to do something right.");
	}
	
	const finalPDF = PDFLib.PDFDocumentFactory.load(arrBuffers.shift());

	while(arrBuffers.length)
	{
		const extraPDF = PDFLib.PDFDocumentFactory.load(arrBuffers.shift());

		for(const page of extraPDF.getPages())
		{
			finalPDF.addPage(page);
		}
	}

	return await PDFLib.PDFDocumentWriter.saveToBytes(finalPDF);
};
```

**Comment by Hopding** [comment]
Hello @oxygen. I took a look at the [code for page addition](https://github.com/Hopding/pdf-lib/blob/master/src/core/pdf-document/PDFDocument.ts#L137-L158). It seems that while the page object itself is being copied into the document, its content streams and resource objects are not being copied. The `PDFDocument.addPage` and `PDFDocument.insertPage` methods will need to be updated to copy these additional objects if the page being added comes from a different document.

**Comment by Hopding** [comment]
@oxygen I just published a new pre-release: [`0.3.1-rc3`](https://www.npmjs.com/package/pdf-lib/v/0.3.1-rc3). It contains a fix for this issue. Please try it out when you get a chance and let me know if it works for you.

You can install this pre-release with npm:
```
npm install pdf-lib@0.3.1-rc3
```
It's also available on unpkg:
* https://unpkg.com/pdf-lib@0.3.1-rc3/dist/pdf-lib.js
* https://unpkg.com/pdf-lib@0.3.1-rc3/dist/pdf-lib.min.js

Note that I'm still working on refining the the page copying logic to handle some edge cases. Stuff like adding the same page to the destination document multiple times, or modifying the page in the source document after adding it to the destination document. But the version contained in this pre-release should work just fine for your use case (assuming it isn't more complicated than the example you shared). 

(Here's the branch for this fix, if you're interested: https://github.com/Hopding)/pdf-lib/tree/CopyPages.

**Comment by Hopding** [comment]
Version [`0.5.0`](https://www.npmjs.com/package/pdf-lib/v/0.5.0) is now published. It contains a completed implementation of the page copying between documents feature. The full release notes are available [here](https://github.com/Hopding/pdf-lib/releases/tag/v0.5.0).

You can install this new version with npm:
```
npm install pdf-lib@0.5.0
```
It's also available on unpkg:
* https://unpkg.com/pdf-lib@0.5.0/dist/pdf-lib.js
* https://unpkg.com/pdf-lib@0.5.0/dist/pdf-lib.min.js
